### ISSUE TITLE
Close connection

### ISSUE THREAD

**Issue by WilsonHo** [NLPchina/elasticsearch-sql#276]
Hi, excuse me, I used your plugin and I call the sql multi time so I check the elasticsearch connection, it don't close immediately, so how can I close the connection?
Thank you,

**Comment by WilsonHo** [comment]
hello, please help me

**Comment by allwefantasy** [comment]
Why you wanna close the connection between this plugin and elasticsearch?

**Comment by eliranmoyal** [comment]
Also from my experience if you create a client per request it can cause a lot of problems.
Using the same instance of the client (i.e singleton) is the best option for TransportClient

**Comment by WilsonHo** [comment]
@eliranmoyal yes, thank you, I think so. But when I test by using nodejs to request like this
`const request = require('request')
var url = 'http://test.server:9200/_sql?sql=SELECT * FROM demoIndex';
for(let i = 0; i < 10000; i+=1){
  setTimeout(()=>{
    request.get(url, (error, response, body) =>{
      if (error) console.log(error);
    })
  }, 100)
}
`
I checked `netstat -nat` and saw the system use many port and get error  { [Error: read ECONNRESET] code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }. So how can I have just one connection?

Thank you

**Comment by shi-yuan** [comment]
@WilsonHo 
I think you are wrong to use [request](https://github.com/request/request)
have a look at [Why one connection per request](https://github.com/request/request/issues/142)
